chainerx_cc/chainerx/cuda/cuda_device/indexing.cu

```diff
@@ -47,56 +47,38 @@ Axes MakeRollingPermutation(int8_t first_axis, int8_t last_axis, int8_t ndim) {
     return permutation;
 }
 
-template <typename T, typename TIndex>
-__global__ void TakeWrapCudaKernel(
-        IndexableArray<const T> a_iarray,
-        IndexableArray<T> out_iarray,
-        IndexableArray<const TIndex> indices_iarray,
-        Indexer<> a_indexer,
-        Indexer<> out_indexer,
-        Indexer<> indices_indexer,
-        TIndex common_total_size,
-        TIndex axis_dim) {
-    static_assert(std::is_same<TIndex, int64_t>::value || std::is_same<TIndex, int32_t>::value, "");
-    for (auto it = out_indexer.It(blockIdx.x * blockDim.x + threadIdx.x, blockDim.x * gridDim.x); it; ++it) {
-        TIndex indices_pos = static_cast<TIndex>(it.raw_index()) / common_total_size;
-        TIndex common_pos = static_cast<TIndex>(it.raw_index()) % common_total_size;
-
-        TIndex index = indices_iarray[indices_indexer.It(indices_pos)];
+template <typename T, typename TIndex, int8_t kNdim>
+__global__ void TakeCudaKernel(
+        IndexableArray<const T, kNdim> x,
+        IndexableArray<const TIndex, kNdim> indices,
+        IndexableArray<T, kNdim> y,
+        Indexer<kNdim> a_indexer,
+        Indexer<kNdim> out_indexer,
+        Indexer<kNdim> indices_indexer,
+        int num_indices,
+        int target_dim,
+        int right_dim,
+        int num_iters,
+        IndexBoundsMode mode) {
+    int idx = blockIdx.x * blockDim.x + threadIdx.x;
+    if (idx >= num_iters) return;
+    int i = idx / (num_indices * right_dim);
```

Comment:
My bad, but it'd be better to give a slightly more readable name for them? Perhaps left_idx, indice_idx, and right_idx or something?